nova_vm/src/ecmascript/builtins/text_processing/string_objects/string_prototype.rs

```diff
@@ -733,73 +734,91 @@ impl StringPrototype {
         Ok(String::from_str(agent, substring).into_value())
     }
 
+    /// ### [22.1.3.23 String.prototype.split ( separator, limit )](https://tc39.es/ecma262/multipage/text-processing.html#sec-string.prototype.split)
     fn split(agent: &mut Agent, this_value: Value, args: ArgumentsList) -> JsResult<Value> {
         // 1. Let O be ? RequireObjectCoercible(this value).
         let o = require_object_coercible(agent, this_value)?;
-        // 2. Let S be ? ToString(O).
-        let s = to_string(agent, o)?;
 
-        // 3. Let isRegExp be ? IsRegExp(searchString).
-        // 4. If isRegExp is true, throw a TypeError exception.
-        // TODO
+        // 2. If separator is neither undefined nor null, then
+        let separator = args.get(0);
 
-        // 5. Let separatorStr be ? ToString(separatorString).
-        let separator_value = args.get(0);
+        if matches!(separator, Value::Undefined | Value::Null) {
+            // TODO: get_method(separator, Symbol.split)
+        }
 
-        // 6. If separator is undefined, return an array with this.
-        if let Value::Undefined = separator_value {
-            let slice: [Value; 1] = [this_value];
-            let results = Array::from_slice(agent, &slice);
+        // 3. Let S be ? ToString(O).
+        let s = to_string(agent, o)?;
 
-            return Ok(results.into_value());
-        }
+        // 4. If limit is undefined, lim is u32::MAX - 1.
+        let limit = args.get(0);
+        let lim = match limit {
```

Comment:
**praise:** I've not actually thought about combining spec `if undefined { X } else { Y }` into matches like this. Makes for fairly pretty code and makes it possible to add fast paths pretty trivially. Nice! <3

Reply:
happy to know at least I did something well! :D